rfd/0067-database-access-aws-redis

```diff
@@ -0,0 +1,221 @@
+
+## Details
+
+### In-transit encryption (TLS)
+For security purpose, Teleport should ONLY support databases with in-transit encryption (TLS) enabled.
+
+[Enabling TLS on ElastiCache Redis](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/in-transit-encryption.html#in-transit-encryption-enable) can be a complicated process, and all client applications must be updated at the same time. Therefore, Teleport should NOT automatically enable TLS for existing clusters. Redis servers without TLS are NOT supported, and users are recommended to enable TLS separately in order to enable Teleport database access.
+
+For clusters with TLS enabled, Teleport performs full certificate validation at connection.
+
+### Auth
+Redis can be configured with one the following authentication methods, and only one at a time.
+
+#### Auth - Redis with no auth
+Redis can be configured without any auth method. "default" user is always used in this configuration, and no `auth` command is required upon succesfull connection.
+
+#### Auth - Redis with Redis Auth
+[Redis Auth](https://redis.io/commands/auth/) uses a single token/password that must be shared for all client applications.
+
+Teleport does NOT support Redis configured in this state, and users are recommended to migrate the database to use ACL/RBAC.<sup>[1]</sup>
+
+#### Auth - Redis with ACL
+[Redis ACL](https://redis.io/docs/manual/security/acl/) (also known as RBAC for ElatiCache Redis) is introduced in Redis 6. Both ElatiCache and MemoryDB provide APIs to manage database users and user groups for ACL.
+
+As with all other databases, the database users should be created by "database admins" (by users, not Teleport) to assign desired permissions.
+
+When an user runs `tsh db connect --db-user <database-user> <aws-rediaws-redis-server>`, database agent automatically sends `auth <database-user> <password>` to Redis server upon succesfull connection, to prevent user from entering the password manually.
+
+To provide better security than using a static password, Teleport must take control of these database users and apply periodic password rotation (more details on secret store in later section). Users can separate the database users used by Teleport vs other client applications.
+
+Here is a sample database service config to specify which database users to manage:
+```
+db_service:
+  enabled: "yes"
+
+  aws:
+  # Discovers ElastiCache/MemoryDB Redis databases.
+  - types: ["elasticache", "memorydb"]
+    regions: ["us-west-1"]
+    tags:
+      "vpc-id": "xyz"
+
+  # Manage (e.g. rotate password for) database users matching provided tags.
+  - types: ["elasticache-user", "memorydb-user"]
+    regions: ["us-west-1"]
+    tags:
+      "managed-by-teleport": "true"
+
+  # Statically list database users (as alternative to using tags matching).
+  users:
+  - arn: "arn:aws:elasticache:ca-central-1:123456789000:user:steve-teleport"
```

Comment:
Instead of having to configure this, would Teleport be able to just query the users with some pre-defined label? For example, we can ask users to just set `teleport.dev/managed: true` (or something like this) tag to their Elasticache/MemDB users and have Teleport discover them?

Reply:
I guess my question rather was, why do we need these additional section at all? Instead of just having Teleport query for users with some predefined label. So e.g. if you have an ElastiCache database in us-west-1, can Teleport just search for all users in us-west-1 with "teleport.dev/managed: true" label? The less configuration the better.